Library/Homebrew/language/python.rb

```diff
@@ -139,11 +139,34 @@ def virtualenv_create(venv_root, python = "python", formula = self)
         venv
       end
 
+      def needs_python?(py)
+        requirements.any? do |req|
+          req.name == py && (req.required? || build.with?(req))
+        end ||
+          deps.any? do |dep|
+            dep.name == py && (dep.required? || build.with?(dep))
+          end
+      end
+
       # Helper method for the common case of installing a Python application.
       # Creates a virtualenv in `libexec`, installs all `resource`s defined
       # on the formula, and then installs the formula.
-      def virtualenv_install_with_resources
-        venv = virtualenv_create(libexec)
+      def virtualenv_install_with_resources(options = {})
+        python = options[:python]
+        if python.nil?
+          py2 = needs_python? "python"
+          py3 = needs_python? "python3"
+          if py2 && !py3
+            python = "python"
+          elsif py3 && !py2
+            python = "python3"
+          elsif py2 && py3
+            raise "ambiguous python, please specify"
+          else
+            python = "python"
+          end
```

Comment:
I think the following code is more succinct and manages to be equally understandable:

```ruby
wanted = %w[python python3].select { |py| needs_python?(py) }
raise "ambiguous python, please specify" if wanted.size > 1
python = wanted.first || "python"
```